### PR TITLE
feat: Spy effect is implemented

### PR DESCRIPTION
**Pull request by asternotus**

### CODE REVIEW

File: effects/spy_effect.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def _activate_impl(self) -> bool:
+        self.targets.append(self.game.spy)
+        return True
+
+    def _resolve_impl(self) -> bool:
+        is_spy_triggered, effect = self._is_spy_triggered()
+
+        if is_spy_triggered:
+            self.game.active_player.hp -= 1
```
Comment (elias-gvin) [inline effects/spy_effect.py]: I'd rather use effect.creator than directly hit active player. But it is not principal

File: effects/spy_effect.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            self.game.spy.hp = 0
+
+            effect.deactivate()
+
+            self.user_interaction.show_active_instant(
+                msg.SpyMessages.RESOLVE_LOST_HP)
+
+        if self.game.spy.hp <= 0:
```
Comment (elias-gvin) [inline effects/spy_effect.py]: Please, use `is_alive` property for such cases

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: effects/spy_effect.py (hunk 1 above)
Comment (asternotus) [reply effects/spy_effect.py:26]: Done

File: effects/spy_effect.py (hunk 2 above)
Comment (asternotus) [reply effects/spy_effect.py]: Done

File: effects/spy_effect.py (hunk 1 above)
Comment (asternotus) [reply effects/spy_effect.py]: Why creator.
The spy is creator of SpyEffect
But I need reduce hp for ActivePlayer

Not change this logic

Comment (elias-gvin) [reply effects/spy_effect.py]: @asternotus the creator of FOUND Kill\Steal\Stagin effect - not creator of SpyEffect

**Review by elias-gvin — CHANGES_REQUESTED** [review]

**Review by elias-gvin — APPROVED** [review]

**Pull request merged**